Crash in fdroid release

When I installed and opened the app, it keeps on crashing.
Device: SM P555
Achitecture: Armeabi v7a
OS: 7 Nougat

FATAL EXCEPTION: main
Process: rocks.poopjournal.morse, PID: 31522
java.lang.RuntimeException: Unable to start activity ComponentInfo{rocks.poopjournal.morse/rocks.poopjournal.morse.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.ImageView.setOnClickListener(android.view.View$OnClickListener)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3003)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3064)
	at android.app.ActivityThread.-wrap14(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1659)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6823)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.ImageView.setOnClickListener(android.view.View$OnClickListener)' on a null object reference
	at rocks.poopjournal.morse.MainActivity.onCreate(MainActivity.java:392)
	at android.app.Activity.performCreate(Activity.java:6977)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1126)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2946)
	... 9 more

--- a/app/src/main/java/rocks/poopjournal/morse/MainActivity.java
+++ b/app/src/main/java/rocks/poopjournal/morse/MainActivity.java
@@ -46,6 +46,7 @@ import android.widget.RelativeLayout;
 import android.widget.TextView;
 import android.widget.Toast;
 
+import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.core.content.ContextCompat;
 import androidx.core.graphics.drawable.DrawableCompat;
@@ -79,12 +80,16 @@ public class MainActivity extends AppCompatActivity implements Camera.AutoFocusC
     RelativeLayout space;
     RelativeLayout makeInputVisible;
     RelativeLayout backspace;
-    RelativeLayout containerTools;
-    RelativeLayout telegraphContainer;
-    RelativeLayout telegraphAudio;
-    RelativeLayout telegraphFlash;
-    RelativeLayout telegraphKey;
-    ImageView telegraphKeyboard,telegraphFlashIV, telegraphAudioIV;
+
+    // These views are only supported in Portrait mode for now, the corresponding variables should be nullable.
+    @Nullable RelativeLayout containerTools;
+    @Nullable  RelativeLayout telegraphContainer;
+    @Nullable RelativeLayout telegraphAudio;
+    @Nullable RelativeLayout telegraphFlash;
+    @Nullable RelativeLayout telegraphKey;
+    @Nullable ImageView telegraphKeyboard;
+    @Nullable ImageView telegraphFlashIV;
+    @Nullable ImageView telegraphAudioIV;
     boolean visibilityCheck = false;
     ArrayList<String> popularMorse = new ArrayList<>();
     HashMap<String, String> popularMorseConversion = new HashMap<>();
@@ -460,68 +465,21 @@ public class MainActivity extends AppCompatActivity implements Camera.AutoFocusC
         popularMorseConversionText.put(".--.....--.....--....--.----...--.-.---..---.....-", "What hath God wrought");
         popularMorseConversionText.put(".-..--...", "rats");
 
-        telegraphKeyboard.setOnClickListener(view -> hideTelegraphKey());
+        if (telegraphKeyboard != null) {
+            telegraphKeyboard.setOnClickListener(view -> hideTelegraphKey());
+        }
         fullscreen.setOnClickListener(view -> showTelegraphKey());
-        telegraphFlash.setOnClickListener(v -> setFlashSelectedForTelegraph());
-        telegraphAudio.setOnClickListener(v -> setAudioSelectedForTelegraph());
+        if (telegraphFlash != null) {
+            telegraphFlash.setOnClickListener(v -> setFlashSelectedForTelegraph());
+        }
+        if (telegraphAudio != null) {
+            telegraphAudio.setOnClickListener(v -> setAudioSelectedForTelegraph());
+        }
         telegraphPlayer= MediaPlayer.create(MainActivity.this,R.raw.beepflac);
 
-        telegraphKey.setOnTouchListener((v, event) -> {
-            if(event.getAction() == MotionEvent.ACTION_DOWN){
-                if (telegraphSelected==1){
-                    time = System.currentTimeMillis();
-                    telegraphPlayer.start();
-                }
-                else {   time = System.currentTimeMillis();
-                    camera = Camera.open();
-                    turnOn();
-                }
-
-
-                return true;
-            }
-            if(event.getAction() == MotionEvent.ACTION_UP){
-                if (telegraphSelected ==1){
-                    if (System.currentTimeMillis()-time>=200){
-                        telegraphPlayer.pause();
-                        telegraphPlayer.seekTo(0);
-                    }
-                    else {
-                        final Handler handler = new Handler(Looper.getMainLooper());
-                        handler.postDelayed(new Runnable() {
-                            @Override
-                            public void run() {
-                                telegraphPlayer.pause();
-                                telegraphPlayer.seekTo(0);
-                            }
-                        }, 100);
-                    }
-                }
-                else {
-
-                    if (System.currentTimeMillis()-time>=200){
-                        turnOff();
-                        camera.release();
-                        camera = null;
-                    }
-                    else {
-                        final Handler handler = new Handler(Looper.getMainLooper());
-                        handler.postDelayed(new Runnable() {
-                            @Override
-                            public void run() {
-                                turnOff();
-                                camera.release();
-                                camera = null;
-                            }
-                        }, 100);
-                    }
-                }
-
-
-                return true;
-            }
-            return false;
-        });
+        if (telegraphKey != null) {
+            telegraphKey.setOnTouchListener((v, event) -> setKeySelectedForTelegraph(event));
+        }
 
 
         flash.setOnClickListener(view -> {
@@ -1201,6 +1159,10 @@ public class MainActivity extends AppCompatActivity implements Camera.AutoFocusC
 
 
     private void showTelegraphKey(){
+        if (containerTools == null || telegraphContainer == null) {
+            return;
+        }
+
         bottomNavigation.setVisibility(View.GONE);
         containerTools.setVisibility(View.GONE);
         telegraphContainer.setVisibility(View.VISIBLE);
@@ -1211,33 +1173,96 @@ public class MainActivity extends AppCompatActivity implements Camera.AutoFocusC
     }
 
     private void hideTelegraphKey(){
+        if (containerTools == null || telegraphContainer == null) {
+            return;
+        }
+
         bottomNavigation.setVisibility(View.VISIBLE);
         containerTools.setVisibility(View.VISIBLE);
         telegraphContainer.setVisibility(View.GONE);
-
     }
 
     private void setAudioSelectedForTelegraph(){
-        Drawable d = (GradientDrawable)telegraphAudio.getBackground();
-        DrawableCompat.setTint(d, Color.parseColor("#227DD3D8"));
+        Drawable d;
+        if (telegraphAudio != null) {
+            d = (GradientDrawable) telegraphAudio.getBackground();
+            DrawableCompat.setTint(d, Color.parseColor("#227DD3D8"));
+        }
 
-        Drawable d2 = (GradientDrawable)telegraphFlash.getBackground();
-        DrawableCompat.setTint(d2, Color.parseColor("#373945"));
+        Drawable d2;
+        if (telegraphFlash != null) {
+            d2 = (GradientDrawable) telegraphFlash.getBackground();
+            DrawableCompat.setTint(d2, Color.parseColor("#373945"));
+        }
 
-        telegraphFlashIV.setColorFilter(Color.parseColor("#9C9CA4"), PorterDuff.Mode.SRC_IN);
-        telegraphAudioIV.setColorFilter(Color.parseColor("#7DD3D8"), android.graphics.PorterDuff.Mode.SRC_IN);
+        if (telegraphFlashIV != null && telegraphAudioIV != null) {
+            telegraphFlashIV.setColorFilter(Color.parseColor("#9C9CA4"), PorterDuff.Mode.SRC_IN);
+            telegraphAudioIV.setColorFilter(Color.parseColor("#7DD3D8"), android.graphics.PorterDuff.Mode.SRC_IN);
+        }
         telegraphSelected =1;
     }
     private void setFlashSelectedForTelegraph(){
         Log.d("flashselected","yes");
-        Drawable d = (GradientDrawable)telegraphFlash.getBackground();
-        DrawableCompat.setTint(d, Color.parseColor("#227DD3D8"));
+        Drawable d;
+        if (telegraphFlash != null) {
+            d = (GradientDrawable) telegraphFlash.getBackground();
+            DrawableCompat.setTint(d, Color.parseColor("#227DD3D8"));
+        }
 
-        Drawable d2 = (GradientDrawable)telegraphAudio.getBackground();
-        DrawableCompat.setTint(d2, Color.parseColor("#373945"));
+        Drawable d2;
+        if (telegraphAudio != null) {
+            d2 = (GradientDrawable) telegraphAudio.getBackground();
+            DrawableCompat.setTint(d2, Color.parseColor("#373945"));
+        }
 
-        telegraphAudioIV.setColorFilter(Color.parseColor("#9C9CA4"), android.graphics.PorterDuff.Mode.SRC_IN);
-        telegraphFlashIV.setColorFilter(Color.parseColor("#7DD3D8"), android.graphics.PorterDuff.Mode.SRC_IN);
+        if (telegraphFlashIV != null && telegraphAudioIV != null) {
+            telegraphAudioIV.setColorFilter(Color.parseColor("#9C9CA4"), PorterDuff.Mode.SRC_IN);
+            telegraphFlashIV.setColorFilter(Color.parseColor("#7DD3D8"), android.graphics.PorterDuff.Mode.SRC_IN);
+        }
         telegraphSelected =2;
+    }
+
+    private boolean setKeySelectedForTelegraph(MotionEvent event) {
+        if (event.getAction() == MotionEvent.ACTION_DOWN) {
+            if (telegraphSelected == 1) {
+                time = System.currentTimeMillis();
+                telegraphPlayer.start();
+            } else {
+                time = System.currentTimeMillis();
+                camera = Camera.open();
+                turnOn();
+            }
+            return true;
+        }
+        if (event.getAction() == MotionEvent.ACTION_UP) {
+            if (telegraphSelected == 1) {
+                if (System.currentTimeMillis() - time >= 200) {
+                    telegraphPlayer.pause();
+                    telegraphPlayer.seekTo(0);
+                } else {
+                    final Handler handler = new Handler(Looper.getMainLooper());
+                    handler.postDelayed(() -> {
+                        telegraphPlayer.pause();
+                        telegraphPlayer.seekTo(0);
+                    }, 100);
+                }
+            } else {
+
+                if (System.currentTimeMillis() - time >= 200) {
+                    turnOff();
+                    camera.release();
+                    camera = null;
+                } else {
+                    final Handler handler = new Handler(Looper.getMainLooper());
+                    handler.postDelayed(() -> {
+                        turnOff();
+                        camera.release();
+                        camera = null;
+                    }, 100);
+                }
+            }
+            return true;
+        }
+        return false;
     }
 }

